### ISSUE TITLE
Take text objects into account in legend autopositioning.

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#7869]
Legend autopositioning currently does not take text objects into account as can be checked by running
```
plt.plot([0, 1], label="foo"); plt.text(.05, .95, "a"); plt.legend()
```
This is because text objects are not included in [`_auto_legend_data`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/legend.py#L707) (which only serves that purpose, AFAICT).

Adding them should be easy, but extra points if the autoplacement algorithm penalizes overlapping with text *more* than overlapping with other artists (given that it's probably indeed worse in general).

**Comment by RiazCharania** [comment]
Is anyone else currently working on this? If not I'll give it a try.

**Comment by timhoffm** [comment]
Go for it.

**Comment by anntzer** [comment]
I still think this would be nice.

**Comment by saranti** [comment]
I think this was fixed by 87d05aea59a339b54477cf405dadb5a7bb9a8c08

**Comment by anntzer** [comment]
Indeed.
